backend/core/src/listings/listings.service.ts

```diff
@@ -1,23 +1,39 @@
 import { Injectable } from "@nestjs/common"
-import { Listing } from "../entity/Listing"
+import { InjectRepository } from "@nestjs/typeorm"
+import { Repository } from "typeorm"
 import { amiCharts } from "../lib/ami_charts"
 import { transformUnits } from "../lib/unit_transformations"
 import { listingUrlSlug } from "../lib/url_helper"
+import jp from "jsonpath"
+import { ListingsListQueryParams, ListingsListResponse } from "./listings.dto"
+import { Listing } from "../entity/listing.entity"
+
+export enum ListingsResponseStatus {
+  ok = "ok",
+}
 
 @Injectable()
 export class ListingsService {
-  public async findAll() {
-    const listings = await Listing.find({ relations: ["units", "attachments", "preferences"] })
+  constructor(@InjectRepository(Listing) private readonly repo: Repository<Listing>) {}
+
+  public async list(options?: ListingsListQueryParams): Promise<ListingsListResponse> {
+    let listings = await this.repo.find({ relations: ["units", "attachments", "preferences"] })
```

Comment:
Whoops, was this a regression? (changing from `Listing.find`?)

Reply:
Fixed